exercisePPJ_06_I/FunctionPPJ_06_I.java

```diff
@@ -0,0 +1,52 @@
+public class FunctionPPJ_06_I {
+	
+	private double binaryWrt[] = {0,0,0,0,0,0,0,0};
+	private double wrtDouble = 0;
+	
+	public void setFunction(double wrtD) {
+		wrtDouble = wrtD;
+	}
+	
+	public void getFunction() {
+		
+		for(int counter = 7; counter >= 0; counter--) {
+		
+		int exponent = 0;
+		
+		while(wrtDouble > Math.pow(2,exponent)) {
+			exponent++;
+		}
+		
+		if(wrtDouble == Math.pow(2,exponent)) {
+			binaryWrt[exponent] = Math.pow(2,exponent);
+		}
+		else {
+			binaryWrt[exponent-1] = Math.pow(2,(exponent-1));
+			
+			int highestExponent = exponent-1;
+			double rest = wrtDouble - Math.pow(2,highestExponent);
+			
+			wrtDouble = rest;
+			exponent=0;
+		}
+	}
+
+		}
+	public void showFunction() {
+		for(int exponent = 0; exponent <= 7; exponent++) {
+		System.out.println("2^"+exponent+": "+binaryWrt[exponent]);
+		}
+	}
+	
+	public void binaryFunction() {
+		System.out.print("Binary: 0b");
+		for(int exponent = 7; exponent >= 0; exponent--) {
+			if(binaryWrt[exponent] != 0.0) {
+				System.out.print("1");
+			}
+			else {
+				System.out.print("0");
+			}
+		}
+	}
+}
```

Comment:
This comment applies to whole task:

In general I like your initiative very much!
Your way of printing is much more interesting (although not perfect).
BUT: please, be precise in fulfilling definition of task. Print what exercise says. (just for now - your creativity will be much appreciated later on)
:+1:  

Another thing:
Try to use this program with argument `0` (zero). Does it work correctly?

Reply:
@MatDls 
yes, of course, I some beautify this program, i think in this way is better and I learn more skill, but ok, I will try do what I should do in exercise only;
when i use input 0 - program is break. i will fix it